examples/tensorflow-MNIST-generalized.ipynb

```diff
@@ -0,0 +1,283 @@
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "(ds_train, ds_test), ds_info = tfds.load(\n",
+    "    'mnist',\n",
+    "    split=['train', 'test'],\n",
+    "    shuffle_files=True,\n",
+    "    as_supervised=True,\n",
+    "    with_info=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def normalize_img(image, label):\n",
+    "  \"\"\"Normalizes images: `uint8` -> `float32`.\"\"\"\n",
+    "  return tf.cast(image, tf.float32) / 255., label\n",
+    "\n",
+    "ds_train = ds_train.map(\n",
+    "    normalize_img, num_parallel_calls=tf.data.experimental.AUTOTUNE)\n",
+    "ds_train = ds_train.cache()\n",
+    "ds_train = ds_train.shuffle(ds_info.splits['train'].num_examples)\n",
+    "ds_train = ds_train.batch(128)\n",
+    "ds_train = ds_train.prefetch(tf.data.experimental.AUTOTUNE)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ds_test = ds_test.map(\n",
+    "    normalize_img, num_parallel_calls=tf.data.experimental.AUTOTUNE)\n",
+    "ds_test = ds_test.batch(128)\n",
+    "ds_test = ds_test.cache()\n",
+    "ds_test = ds_test.prefetch(tf.data.experimental.AUTOTUNE)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Epoch 1/3\n",
+      "469/469 [==============================] - 1s 2ms/step - loss: 0.3558 - accuracy: 0.9021 - val_loss: 0.1865 - val_accuracy: 0.9458\n",
+      "Epoch 2/3\n",
+      "469/469 [==============================] - 1s 1ms/step - loss: 0.1606 - accuracy: 0.9536 - val_loss: 0.1407 - val_accuracy: 0.9575\n",
+      "Epoch 3/3\n",
+      "469/469 [==============================] - 1s 1ms/step - loss: 0.1157 - accuracy: 0.9668 - val_loss: 0.1060 - val_accuracy: 0.9679\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "<tensorflow.python.keras.callbacks.History at 0x7fb1cc04c668>"
+      ]
+     },
+     "execution_count": 20,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "model = tf.keras.models.Sequential([\n",
+    "  tf.keras.layers.Flatten(input_shape=(28, 28, 1)),\n",
+    "  tf.keras.layers.Dense(128,activation='relu'),\n",
+    "  tf.keras.layers.Dense(10, activation='softmax')\n",
+    "])\n",
+    "model.compile(\n",
+    "    loss=tf.keras.losses.sparse_categorical_crossentropy,\n",
+    "    optimizer=tf.keras.optimizers.Adam(0.001),\n",
+    "    metrics=['accuracy'],\n",
+    ")\n",
+    "\n",
+    "model.fit(\n",
+    "    ds_train,\n",
+    "    epochs=3,\n",
+    "    validation_data=ds_test,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class BCCrossEntropyLoss(object):\n",
```

Comment:
Is there a reason why you chose to copy-paste this class instead of importing it?

Reply:
Good catch. This was leftover.

Reply:
Fixed and pushed changes.